client/src/components/TaskDetails.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+
+// Mapping between database column names and display subheading titles to reduce further data processesing
+const sectionNamesMap: { [key: string]: string } = {
+  category: "Task Category",
+  title: "Task Title",
+  created_at: "Date",
+  suburb: "Suburb",
+  state: "State",
+  estimated_time: "Estimated Time", // Missing ERD column name (fix later when db added)
+  budget: "Estimated Price",
+  description: "Task Details",
+};
+
+// Type of the section data to be passed
+type SectionData = {
+  [key: string]: string | null; // Allow database column to have null for item (prints 'N/A')
+};
+
+// Props for TaskDetails (and the sections)
+type TaskDetailsProps = {
+  data: SectionData;
+  emptyContentPlaceholder?: string; // OPTIONAL: What to show if missing content for a subheading.
+};
+
+const TaskDetailsSection = ({
+  data,
+  emptyContentPlaceholder = "N/A", // Change this to modify default placeholder
+}: TaskDetailsProps) => {
+  return (
+    <>
+      {Object.keys(sectionNamesMap).map((dbName, index) => (
+        <div
+          key={index}
+          className="py-2.5 text-base font-normal leading-5 text-[#485358]"
+        >
+          {sectionNamesMap[dbName]}
+          <p className="py-1.5 text-xl font-normal leading-7 text-[#000000]">
```

Comment:
Same goes for here, replace with colours defined in tailwind config file (don't think text is actually pure black). And replace `text-xl font-normal leading-7` with styles in `Styles.css`

Reply:
in the figma design 'Task Details' is #000000 but the rest of the dark text is the regular light text, it has been updated based off this